confidential/confidential.go

```diff
@@ -91,3 +90,63 @@ func FinalValueBlindingFactor(input FinalValueBlindingFactorInput) ([32]byte, er
 
 	return secp256k1.BlindGeneratorBlindSum(ctx, values, generatorBlind, blindingFactor, len(input.InValues))
 }
+
+type RangeProofInput struct {
+	Value               uint64
+	BlindingPubkey      []byte
+	EphemeralPrivkey    []byte
+	Asset               []byte
+	AssetBlindingFactor []byte
+	ValueBlindFactor    [32]byte
+	ValueCommit         []byte
+	ScriptPubkey        []byte
+	MinValue            uint64
+	Exp                 int
+	MinBits             int
+}
+
+//RangeProof method calculates range proof
+func RangeProof(input RangeProofInput) ([]byte, error) {
+	ctx, _ := secp256k1.ContextCreate(secp256k1.ContextBoth)
+	defer secp256k1.ContextDestroy(ctx)
+
+	nonce, err := NonceHash(ctx, input.BlindingPubkey, input.EphemeralPrivkey)
+	if err != nil {
+		return nil, err
+	}
+
+	generator, err := secp256k1.GeneratorGenerateBlinded(ctx, input.Asset, input.AssetBlindingFactor)
+	if err != nil {
+		return nil, err
+	}
+
+	message := append(input.Asset, input.AssetBlindingFactor...)
+
+	commit, err := secp256k1.CommitmentParse(ctx, input.ValueCommit)
```

Comment:
You should use the `CommitmentFromBytes` factory method that wraps the binding you actually used:
```suggestion
	commit, err := secp256k1.CommitmentFromBytes(ctx, input.ValueCommit)
```